packages/cli/src/NodeTypes.ts

```diff
@@ -42,15 +48,17 @@ export class NodeTypes implements INodeTypes {
 		return this.getNode(nodeType).type;
 	}
 
-	getByNameAndVersion(nodeType: string, version?: number): INodeType {
+	getByNameAndVersion(nodeType: string, version?: number): INodeType | undefined {
```

Comment:
1. Thanks for challenging this. After looking at this, it seems `getByName` is only used in one place. Will just get rid of it and use `getByNameAndVersion` directly
2. Not really. See this function `getVersionedNodeType` here it's calling https://github.com/n8n-io/n8n/blob/3b01eb60c98d51d0d7572342b8d6d40763293719/packages/workflow/src/NodeHelpers.ts#L1588
3. In NodeTypes, it would be coming from `getNodeType` with an unknown version https://github.com/n8n-io/n8n/blob/3b01eb60c98d51d0d7572342b8d6d40763293719/packages/workflow/src/VersionedNodeType.ts#L23
I was doing a unit test for an unknown version in another PR but maybe that's not a realistic case.